From 10ec20a927556eb12a111c9e5c45e5103fd94a98 Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Wed, 27 Jun 2018 13:24:56 +0800
Subject: [PATCH 1/4] Add to support the VehicleProperty HVAC_SEAT_TEMPERATURE

Fix the crash of HVAC application when tap the seat temperature button.

Change-Id: I86f911d12cf520f63834363a09fc9f029f6fbeaa
Jira: OAM-65772
Test: Refer to the steps from JIRA
Signed-off-by: Lei,RayX <rayx.lei@intel.com>
---
 .../vehicle/2.0/default/impl/vhal_v2_0/DefaultConfig.h  | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/automotive/vehicle/2.0/default/impl/vhal_v2_0/DefaultConfig.h b/automotive/vehicle/2.0/default/impl/vhal_v2_0/DefaultConfig.h
index 08d3d79..5c0b994 100644
--- a/automotive/vehicle/2.0/default/impl/vhal_v2_0/DefaultConfig.h
+++ b/automotive/vehicle/2.0/default/impl/vhal_v2_0/DefaultConfig.h
@@ -211,6 +211,23 @@ const ConfigDeclaration kVehicleProperties[]{
      .initialAreaValues = {{toInt(VehicleAreaZone::ROW_1_LEFT), {.floatValues = {16}}},
                            {toInt(VehicleAreaZone::ROW_1_RIGHT), {.floatValues = {20}}}}},
 
+    {.config = {.prop = toInt(VehicleProperty::HVAC_SEAT_TEMPERATURE),
+                .access = VehiclePropertyAccess::READ_WRITE,
+                .changeMode = VehiclePropertyChangeMode::ON_CHANGE,
+                .supportedAreas = VehicleAreaZone::ROW_1_LEFT | VehicleAreaZone::ROW_1_RIGHT,
+                .areaConfigs = {VehicleAreaConfig{
+                                    .areaId = toInt(VehicleAreaZone::ROW_1_LEFT),
+                                    .minInt32Value = 0,
+                                    .maxInt32Value = 3,
+                                },
+                                VehicleAreaConfig{
+                                    .areaId = toInt(VehicleAreaZone::ROW_1_RIGHT),
+                                    .minInt32Value = 0,
+                                    .maxInt32Value = 3,
+                                }}},
+     .initialAreaValues = {{toInt(VehicleAreaZone::ROW_1_LEFT), {.int32Values = {0}}},
+                           {toInt(VehicleAreaZone::ROW_1_RIGHT), {.int32Values = {0}}}}},
+
     {.config =
          {
              .prop = toInt(VehicleProperty::ENV_OUTSIDE_TEMPERATURE),
-- 
1.9.1

